src/ring_theory/jacobson.lean

```diff
@@ -535,60 +582,50 @@ lemma is_jacobson_mv_polynomial_fin [H : is_jacobson R] :
+variables {n : ℕ}
+
+lemma quotient_mk_comp_C_is_integral_of_jacobson {R : Type*} [integral_domain R] [is_jacobson R]
+  (P : ideal (mv_polynomial (fin n) R)) [P.is_maximal] :
   ((quotient.mk P).comp mv_polynomial.C : R →+* P.quotient).is_integral :=
 begin
   unfreezingI {induction n with n IH},
-  { have := (ring_equiv_of_equiv R (equiv.equiv_pempty $ fin.elim0)).trans (pempty_ring_equiv R),
-    refine ring_hom.is_integral_of_surjective _ (function.surjective.comp quotient.mk_surjective _),
+  { refine ring_hom.is_integral_of_surjective _ (function.surjective.comp quotient.mk_surjective _),
     exact C_surjective_fin_0 },
-  { let ϕ1 : R →+* mv_polynomial (fin n) R := mv_polynomial.C,
-    let ϕ2 : (mv_polynomial (fin n) R) →+* polynomial (mv_polynomial (fin n) R) := polynomial.C,
-    let ϕ3 := (fin_succ_equiv R n).symm.to_ring_hom,
-    let ϕ : R →+* (mv_polynomial (fin (n+1)) R) := ϕ3.comp (ϕ2.comp ϕ1),
-
-    let P3 : ideal (polynomial (mv_polynomial (fin n) R)) := P.comap ϕ3,
-    let P2 : ideal (mv_polynomial (fin n) R) := P3.comap ϕ2,
-    let P1 : ideal R := P2.comap ϕ1,
-    haveI : P3.is_maximal := comap_is_maximal_of_surjective ϕ3 (fin_succ_equiv R n).symm.surjective,
-    haveI : P2.is_maximal := polynomial.is_maximal_comap_C_of_is_jacobson P3,
-
-    let φ3 : P3.quotient →+* P.quotient := quotient_map P ϕ3 le_rfl,
-    let φ2 : P2.quotient →+* P3.quotient := quotient_map P3 ϕ2 le_rfl,
-    let φ1 : P1.quotient →+* P2.quotient := quotient_map P2 ϕ1 le_rfl,
-    let φ : P1.quotient →+* P.quotient := φ3.comp (φ2.comp φ1),
-
-    have hφ3 : φ3.is_integral := φ3.is_integral_of_surjective
-      (quotient_map_surjective (fin_succ_equiv R n).symm.surjective),
-    have hφ2 : φ2.is_integral,
-    { rw is_integral_quotient_map_iff ϕ2,
-      refine polynomial.quotient_mk_comp_C_is_integral_of_jacobson P3 } ,
-    have hφ1 : φ1.is_integral,
-    { rw is_integral_quotient_map_iff ϕ1,
-      refine IH P2 },
-    have hφ : φ.is_integral := ring_hom.is_integral_trans (φ2.comp φ1) φ3
-      (ring_hom.is_integral_trans φ1 φ2 hφ1 hφ2) hφ3,
-
-    have : (quotient.mk P).comp ϕ = φ.comp (quotient.mk P1),
-    by rw [ring_hom.comp_assoc, ring_hom.comp_assoc, quotient_map_comp_mk, ← ring_hom.comp_assoc ϕ1,
-      quotient_map_comp_mk, ← ring_hom.comp_assoc, ← ring_hom.comp_assoc, ← ring_hom.comp_assoc,
-      ← ring_hom.comp_assoc, quotient_map_comp_mk],
-    rw [← fin_succ_equiv_comp_C_eq_C n, this],
-    refine ring_hom.is_integral_trans (quotient.mk P1) φ _ hφ,
-    exact (quotient.mk P1).is_integral_of_surjective (quotient.mk_surjective) }
+  { rw [← fin_succ_equiv_comp_C_eq_C, ← ring_hom.comp_assoc, ← ring_hom.comp_assoc,
+      ← @quotient_map_comp_mk _ _ _ _ _ _ _ le_rfl, ring_hom.comp_assoc (polynomial.C),
+      ← @quotient_map_comp_mk _ _ _ _ _ _ _ le_rfl, ring_hom.comp_assoc, ring_hom.comp_assoc,
+      ← @quotient_map_comp_mk _ _ _ _ _ _ _ le_rfl, ← ring_hom.comp_assoc (quotient.mk _)],
```

Comment:
Do you think `quotient_map_comp_mk` should be defined with more explicit arguments to avoid the `@` here?